react/features/device-selection/functions.js

```diff
@@ -26,20 +26,30 @@ import { toState } from '../base/redux';
 export function getDeviceSelectionDialogProps(stateful: Object | Function) {
     const state = toState(stateful);
     const settings = state['features/base/settings'];
+    const { conference } = state['features/base/conference'];
+    let disableAudioInputChange = !JitsiMeetJS.mediaDevices.isMultipleAudioInputSupported();
+
+    // multiple audio input change will be a problem only when we are in a
+    // conference, when we open device selection on welcome page
+    // changing input devices will not be a problem
+    if (!conference) {
+        disableAudioInputChange = false;
+    }
 
+    // we fill the device selection dialog with the devices that are currently
+    // used or if none are currently used with what we have in settings(user selected)
     return {
         availableDevices: state['features/base/devices'].availableDevices,
-        disableAudioInputChange:
-            !JitsiMeetJS.isMultipleAudioInputSupported(),
+        disableAudioInputChange,
         disableDeviceChange:
             !JitsiMeetJS.mediaDevices.isDeviceChangeAvailable(),
         hideAudioInputPreview:
             !JitsiMeetJS.isCollectingLocalStats(),
         hideAudioOutputSelect: !JitsiMeetJS.mediaDevices
                             .isDeviceChangeAvailable('output'),
-        selectedAudioInputId: settings.micDeviceId,
-        selectedAudioOutputId: getAudioOutputDeviceId(),
-        selectedVideoInputId: settings.cameraDeviceId
+        selectedAudioInputId: settings.micDeviceId || settings.userSelectedMicDeviceId,
```

Comment:
Do we really need to fallback to the user selected values? Do we have a use case when we need to do this? 

I'm wondering about the use case when you don't have a device but you have something stored from another session. Isn't this going to cause showing the user selected device from previous session as currently used?

Reply:
Updated so we distinguish between welcome page settings and in conference settings.